packages/hub/src/app/new/model/NewModel.tsx

```diff
@@ -28,35 +28,55 @@ const Mutation = graphql`
   }
 `;
 
+const defaultCode = `/*
+Describe your code here
+*/
+
+a = normal(2, 5)
+`;
+
 export const NewModel: FC = () => {
   const { data: session } = useSession({ required: true });
 
   const toast = useToast();
 
   const form = useForm<{
-    code: string;
-    slug: string;
+    slug: string | undefined;
   }>({
     defaultValues: {
-      code: `/*
-Describe your code here
-*/
-
-a = normal(2, 5)`,
+      // don't pass `slug: ""` here, it will lead to form reset if a user started to type in a value before JS finished loading
     },
+    mode: "onChange",
   });
 
+  const watchSlug = form.watch("slug");
+
+  useEffect(() => {
+    // watchSlug can be undefined if the page was just loaded
+    if (watchSlug && watchSlug.includes(" ")) {
```

Comment:
I assumed this would be done by watching for key presses, but I guess this works too

Reply:
I agree it's weird, but it's hard to intercept key presses immediately, because `<TextFormField>` is an uncontrolled component.

I could use `<ControlledFormField>` instead, maybe I should've, not sure, but it would be complicated in a different way.

Reply:
Makes sense. When I tried doing this (very briefly), this was an issue.
Controlled components are probably better, but I'm fine with this shortcut here.